scipy/signal/tests/test_signaltools.py

```diff
@@ -1093,20 +1093,13 @@ def test_types(self, dtype):
         assert_equal(signal.medfilt(in_typed).dtype, dtype)
         assert_equal(signal.medfilt2d(in_typed).dtype, dtype)
 
-    def test_types_deprecated(self):
-        dtype = np.longdouble
-        in_typed = np.array(self.IN, dtype=dtype)
-        msg = "Using medfilt with arrays of dtype"
-        with pytest.deprecated_call(match=msg):
-            assert_equal(signal.medfilt(in_typed).dtype, dtype)
-        with pytest.deprecated_call(match=msg):
-            assert_equal(signal.medfilt2d(in_typed).dtype, dtype)
-
-
     @pytest.mark.parametrize('dtype', [np.bool_, np.complex64, np.complex128,
-                                       np.clongdouble, np.float16,])
+                                       np.clongdouble, np.float16, np.longdouble,
+                                       np.object_])
     def test_invalid_dtypes(self, dtype):
         in_typed = np.array(self.IN, dtype=dtype)
```

Comment:
```suggestion
                                       np.clongdouble, np.float16,
                                       np.object_, "float96", "float128"])
    def test_invalid_dtypes(self, dtype):
        if system_does_not_have_large_float and dtype in ["float96", "float128"]:
            pytest.mark.skip("Platform does not support `float128`")
        in_typed = np.array(self.IN, dtype=dtype)
```

`system_does_not_have_large_float` is left as an exercise to the reader. ;-)